database/spanner/spanner.go

```diff
@@ -316,3 +341,18 @@ func migrationStatements(migration []byte) []string {
 	}
 	return nonEmptyStatements
 }
+
+func cleanStatements(migration []byte) ([]string, error) {
+	// The Spanner GCP backend does not yet support comments for the UpdateDatabaseDdl RPC
+	// (see https://issuetracker.google.com/issues/159730604) we use
+	// spansql to parse the DDL and output valid stamements without comments
+	ddl, err := spansql.ParseDDL("", string(migration))
+	if err != nil {
+		return nil, err
+	}
+	stmts := []string{}
```

Comment:
Nit: pre-allocate the slice e.g. `make([]string, 0, len(ddl.List))`

Reply:
Done